### PR TITLE
manylinux llvmdev upgrade to LLVM 11 / manylinux2010 to manylinux2014 upgrade

### PR DESCRIPTION
**Pull request by esc**
Getting the new llvmdev wheels going. I may fold the manylinux2010 -> manylinux2014 upgrade into this PR too.

### CODE REVIEW

**Comment by esc** [comment]
With the latest changes on this PR I was able to build `llvmdev` packages for linux-32 and linux-64:

https://anaconda.org/numba/llvmdev/files?channel=manylinux2014

**Comment by esc** [comment]
and now for `linux-aarch64` too:

https://anaconda.org/numba/llvmdev/files?channel=manylinux2014

**Comment by esc** [comment]
Unfortunately when trying to build the llvmlite wheel for aarch64 I got:

```
 INFO - INFO:auditwheel.main_repair:Repairing llvmlite-0.37.0rc1+7.g3b8129f-cp38-cp38-linux_aarch64.whl
 INFO - usage: auditwheel [-h] [-V] [-v] command ...
 INFO - auditwheel: error: cannot repair "llvmlite-0.37.0rc1+7.g3b8129f-cp38-cp38-linux_aarch64.whl" to "manylinux2014_aarch64" ABI because of the presence of too-recent versioned symbols. You'll need to compile the wheel on an older toolchain.
```

**Comment by esc** [comment]
I think this is now ready for review.

File: buildscripts/manylinux2014/README.md [hunk 1]
```diff
@@ -31,6 +32,7 @@ Available Python installations (`<pyver>`) are:
 - cp36-cp36m
 - cp37-cp37m
 - cp38-cp38
+- cp39-cp39
```
Comment (hrw) [inline buildscripts/manylinux2014/README.md:36]: - cp310-cp310

There is Python 3.10 in manylinux2014 as well (beta).

File: buildscripts/manylinux2014/configure_conda.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Setup miniconda environment that is compatible with manylinux2014 docker image
```
Comment (hrw) [inline buildscripts/manylinux2014/configure_conda.sh:1]: add #!/bin/bash as [[ is bashism.

File: buildscripts/manylinux2014/README.md (hunk 1 above)
Comment (esc) [reply buildscripts/manylinux2014/README.md:36]: Thank you for the info, I don't think we will want to build Python 3.10 for the time being. Most likely for the next release.

File: buildscripts/manylinux2014/configure_conda.sh (hunk 2 above)
Comment (esc) [reply buildscripts/manylinux2014/configure_conda.sh:2]: done in 2f0a24af7e3b975819346638a146046faa82784e

**Comment by sklam** [comment]
i tested the linux wheels on a fresh py3.9 conda env using:

```
pip install --extra-index https://pypi.anaconda.org/numba/label/manylinux2014_testing_wheels/simple numba==0.54.0rc1
python -m numba.runtests --random=0.1 -m -vb
```
and it looks good.

File: conda-recipes/llvmdev_manylinux2014/meta.yaml [hunk 3]
```diff
@@ -1,21 +1,8 @@
...
-{% set sha256_llvm = "c5d8e30b57cbded7128d78e5e8dad811bff97a8d471896812f57fa99ee82cdf3" %}
-{% set sha256_lld = "591449e0aa623a6318d5ce2371860401653c48bb540982ccdd933992cb88df7a" %}
-{% set build_number = "1" %}
-
-{% endif %}
+{% set shortversion = "11.1" %}
+{% set version = "11.1.0" %}
+{% set sha256_llvm = "ce8508e318a01a63d4e8b3090ab2ded3c598a50258cc49e2625b9120d4c03ea5" %}
+{% set sha256_lld = "017a788cbe1ecc4a949abf10755870519086d058a2e99f438829aef24f0c66ce" %}
+{% set build_number = "2" %}
```
Comment (sklam) [inline conda-recipes/llvmdev_manylinux2014/meta.yaml:5]: why build_number 2?

**Review by sklam — COMMENTED** [review]
The patch looks good. I have just one non-critical comment.

File: conda-recipes/llvmdev_manylinux2014/meta.yaml (hunk 3 above)
Comment (esc) [reply conda-recipes/llvmdev_manylinux2014/meta.yaml:5]: This is historical, the other conda `llvmdev` (not the manylinux ones) also has build number two, so for coherence this build number was adapted here too.

**Review by sklam — APPROVED** [review]
Thanks for the patch!

**Pull request merged**